src/reducers/notebook-reducer.js

```diff
@@ -210,7 +209,11 @@ const notebookReducer = (state = newNotebook(), action) => {
 
     case "LOGIN_SUCCESS": {
       const { userData } = action;
-      return Object.assign({}, state, { userData });
+      const { notebookInfo } = state;
+      if (userData.name === state.notebookInfo.username) {
+        notebookInfo.user_can_save = true;
+      }
+      return Object.assign({}, state, { userData }, { notebookInfo });
```

Comment:
I wonder if this could be written more concisely as:
```js
return Object.assign({}, state, { userData }, { notebookInfo: { ...state.notebookInfo, user_can_save: userData.name === state.notebookInfo.username });
```